### ISSUE TITLE
Couple of thoughts

### ISSUE THREAD

**Issue by mxinden** [thomaseizinger/semverlog#1]
Thanks for the detailed README!

- What is the use-case for `priority`?

- > For example, we don't integrate with package managers
  
   Is the gained flexibility worth the additional complexity? In case `semverlog` would integrate with the Rust ecosystem `semverlog compute-bump-level` would not require the current version as an argument, correct? Also `semverlog compile-changelog` would not require the new-version to be provided as it can derive it from `semverlog compute-bump-level` and the current `Cargo.toml`.

- Using a separate file for each change to avoid merge conflicts is neat!
- In the case of rust-libp2p, would we have one `.changes/` directory per crate?
- Nit: Consider automatically deleting the files in `.changes/` after generating the changelog.

**Comment by thomaseizinger** [comment]
> * > For example, we don't integrate with package managers
>       
>       
>       Is the gained flexibility worth the additional complexity? In case `semverlog` would integrate with the Rust ecosystem `semverlog compute-bump-level` would not require the current version as an argument, correct? Also `semverlog compile-changelog` would not require the new-version to be provided as it can derive it from `semverlog compute-bump-level` and the current `Cargo.toml`.

I recently had the idea of creating a separate `cargo semverlog` integration to address this.

> * What is the use-case for `priority`?

Sorting changelog entries within the final changelog.

> * In the case of rust-libp2p, would we have one `.changes/` directory per crate?

Yes.

> * Nit: Consider automatically deleting the files in `.changes/` after generating the changelog.

I guess because we are integrated with git, this is safe? I'll think about it.

> * Using a separate file for each change to avoid merge conflicts is neat!

Not only that, it also prevents merging a changelog entry into an already released version. Something that has happened several times for us already.

**Comment by mxinden** [comment]
> > * Nit: Consider automatically deleting the files in `.changes/` after generating the changelog.
> 
> I guess because we are integrated with git, this is safe? I'll think about it.

Yes. Also worth considering is failing early in case the directory is in a (git) dirty state.
